golang/cosmos/x/vtransfer/ibc_middleware_test.go

```diff
@@ -314,139 +382,298 @@ func (s *IntegrationTestSuite) mintToAddress(chain *ibctesting.TestChain, addr s
 	s.Require().NoError(err)
 	err = app.BankKeeper.SendCoinsFromModuleToAccount(chain.GetContext(), ibctransfertypes.ModuleName, addr, coins)
 	s.Require().NoError(err)
-
-	// Verify success.
-	balances := app.BankKeeper.GetAllBalances(chain.GetContext(), addr)
-	s.Require().Equal(coins[0], balances[1])
 }
 
-// TestTransferFromAgdToAgd relays an IBC transfer initiated from a chain A to a
-// chain B, and relays the chain B's resulting acknowledgement in return. It
-// verifies that the source and destination accounts' bridge targets are called
-// by inspecting their resulting actionQueue records.  By committing blocks
-// between actions, the test verifies that the VM results are permitted to be
-// async across blocks.
-func (s *IntegrationTestSuite) TestTransferFromAgdToAgd() {
-	path := s.NewTransferPath()
-	s.Require().Equal(path.EndpointA.ChannelID, "channel-1050")
-
-	s.Run("TransferFromAgdToAgd", func() {
-		// create a transfer packet's data contents
-		baseReceiver := s.chainB.SenderAccounts[1].SenderAccount.GetAddress().String()
-		receiverHook, err := types.JoinHookedAddress(baseReceiver, []byte("?what=arbitrary-data&why=to-test-bridge-targets"))
-		s.Require().NoError(err)
-		transferData := ibctransfertypes.NewFungibleTokenPacketData(
-			"uosmo",
-			"1000000",
-			s.chainA.SenderAccount.GetAddress().String(),
-			receiverHook,
-			`"This is a JSON memo"`,
-		)
+// TestHops relays an IBC transfer initiated from a chain A to a chain B, via 0
+// or more intermediate chains' PacketForwardMiddleware, and relays the chain
+// B's resulting acknowledgement back through the intermediate chains to chain A
+// in return. It verifies that the source and destination accounts' bridge
+// targets are called by inspecting their resulting actionQueue records.  By
+// committing blocks between actions, the test verifies that the VM results are
+// permitted to be async across blocks.
+func (s *IntegrationTestSuite) TestHops() {
```

Comment:
Is there any way to test after completion of a round trip that all chains have pruned all `originalData` state it would have temporarily created?